### DISCUSSION TITLE
How to set the compilation setting of calling Python code in OMEdit?

### DISCUSSION THREAD

**Discussion by Amuri44** [OpenModelica/OpenModelica#8574] [Q&A]
I am reading section 22.2 of the OpenModelica User Guide which looks helpful. However, I came a cross this part
![Screenshot 2022-02-17 184503](https://user-images.githubusercontent.com/86628555/154592511-0decf1c0-49c6-4e6c-889f-66222495d560.png)
In which I dont know how to define these settings in the OMEdit rather than the standard Python API. Any idea for that?

**Comment by AnHeuermann** [comment]
If you are using the scripting API you can use [`setLinkerFlags()`](https://openmodelica.org/doc/OpenModelicaUsersGuide/latest/scripting_api.html#setlinkerflags) as well.

From OMEdit you can add C flags in the Simulation Setup in the General tap in `C/C++ Compiler Flags (Optional)`:
![grafik](https://user-images.githubusercontent.com/38031952/214833824-ac947fc8-11f2-401e-bb21-28238e996699.png)

These flags will be appended to the usual C flags. So they won't replace the usually used flags.
I don't think there is a way to replace the C or linker flags from OMEdit directly.

**Reply by Amuri44** [reply]
So, the only way to call/simulate python code, is to utilize Python API?
Is there so far an API for Julia language?

**Reply by AnHeuermann** [reply]
> So, the only way to call/simulate python code, is to utilize Python API?

No. If I get your use case right you are trying to call some Python code inside a Modelica model. And for that you are adding Python linker and compiler flags to the OpenModelica compiler. In that case it doesn't matter how you call OpenModelica as long as you are providing the correct flags.

But a better approach would probably be to create a C wrapper library for your python code and use the [external function interface](https://specification.modelica.org/maint/3.5/functions.html#external-function-interface) of the Modelica language. With that your solution should be working with every Modelica tool, not only OpenModelica.

> Is there so far an API for Julia language?

There are 4 API's to use OpenModelica for scripting from different languages:

- [Scripting API](https://openmodelica.org/doc/OpenModelicaUsersGuide/latest/scripting_api.html)
- Julia interface [OMJulia](https://github.com/OpenModelica/OMJulia.jl)
- Python interface [OMPython](https://github.com/OpenModelica/OMPython)
- Matlab interface [OMMatlab](https://github.com/OpenModelica/OMMatlab)
